unit-tests/native/src/test/scala/org/scalanative/testsuite/javalib/lang/ref/WeakReferenceTest.scala

```diff
@@ -81,6 +79,12 @@ class WeakReferenceTest {
     val weakRef1 = allocWeakRef(refQueue, new WeakReference(_, _))
     val weakRef2 = allocWeakRef(refQueue, new WeakReference(_, _))
     val weakRef3 = allocWeakRef(refQueue, new SubclassedWeakRef(_, _))
+    var ref4 = A()
+    var ref4Collected = false
+    if (Platform.executingInScalaNative)
+      WeakReferenceRegistry.addHandler(ref4) { ref4Collected = true }
```

Comment:
All `.class` or `.tasty` files produced by `javalib`, `scalalib`, `auxlib` are not published, you would not be able to access them outside of that modules. 
You'd need to define a Java source describing their interface (would produce only .class), and actual implementation would be provided from javalib `.nir` files

Reply:
Also the `scalanative.regex.Matcher` is defined in nativelib

Reply:
ah... would it work if we move the entire java.lang.ref package to nativelib, then? or is that not allowed?

if not, where would i put a java file -- and what interface to publish if WeakReferenceRegistry is an object, hence a bunch of static methods?